spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/ContractId.java

```diff
@@ -83,16 +135,20 @@ public String definitionPart() {
      * @return The definition part of the id
      */
     public String assetIdPart() {
-        var parts = parseContractId(value);
-        return parts[ASSET_ID_PART];
-    }
-
-    private String[] parseContractId(@NotNull String id) {
-        return id.split(DELIMITER);
+        return assetId;
     }
 
     @Override
     public String toString() {
-        return value;
+        var encoder = Base64.getEncoder();
+        return encoder.encodeToString(definitionId.getBytes()) +
+                DELIMITER +
+                encoder.encodeToString(assetId.getBytes()) +
+                DELIMITER +
+                encoder.encodeToString(uuid.getBytes());
+    }
+
+    public ContractId spawn() {
```

Comment:
this deserves some documentation, since its purpose is not readily deducible from the name. on that note, maybe a name like `derive` or `copy` would be better?